src/Login.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import { useState, ChangeEvent, FormEvent } from 'react';
+import {
+  Text,
+  Button,
+  TextInput,
+  PasswordInput,
+} from '@mantine/core'
+
+interface FormValues {
```

Comment:
why interface vs type? what are the differences between the two?

Reply:
So definitely right high level idea but slight correction. Types could actually be extended but you're not 'extending' as much as you are composing. For example `type c = a & b`. This type logic where you can calculate intersections and unions is actually very powerful for complex data types. Most of the time though, I would just stick with one or the other and it comes down to personal preference. I prefer types but recognize that interfaces make things cleaner and offers better inspection for more niche linting too.

Reply:
Got it I think, I suppose if just declaring anything other than an object, might as well use a type, and for objects and declarations that are subject to possible change down the line, go with interface.